pkgs/development/compilers/ghcjs/base.nix

```diff
@@ -159,6 +163,7 @@ in mkDerivation (rec {
         --with-cabal ${cabal-install}/bin/cabal \
         --with-gmp-includes ${gmp.dev}/include \
         --with-gmp-libraries ${gmp.out}/lib
+      ${if isNull ghcLibdir then "" else ''echo "${ghcLibdir}" > "$out/lib/ghcjs-${version}/ghc_libdir"''}
```

Comment:
@cstrahan, @ryantrinkle briefly explained to me that GHCJS has a "hack" that checks for the presence of this symlink and will use it as a way of adding GHCJS plugins. @ryantrinkle uses a plugin in reflex platform that is designed to add more performance optimizations. It is minimally effective at this point but he plans to expand it extensively.

However, for any sort of real accuracy or precision, we would need to hear directly from @ryantrinkle or @luite.